app/error_handlers.py

```diff
@@ -40,6 +42,11 @@ def create_error_middleware(overrides):
     return middleware_handler
 
 
+async def ce_closed(request):
+    logger.info("Attempt to access collection exercise that has already ended")
```

Comment:
Would it be worth logging the collection exercise ID here, or will that mean passing a collex ID around between functions unnecessarily?

Reply:
It would be useful but I'm not sure we can easily pass that around without some refactoring.

Reply:
Nice work!